INSTALL.md

```diff
@@ -0,0 +1,136 @@
+library install, don't mind getting head (which may be less stable than a
+particular release) and don't share any dependencies with client-go, then
+simply:
+
+```sh
+$ go get k8s.io/client-go/...
+
+# Test install
+$ go build k8s.io/client-go/examples/...
+```
+
+This will install `k8s.io/client-go` in your `$GOPATH`. `k8s.io/client-go` includes its own
+dependencies in its `k8s.io/client-go/vendor` path. `go get` will not flatten
+those dependencies into your `$GOPATH`, which means they will be distinct from
+any dependencies you may already have there. This will be problematic if you
+happen to already have a copy of, say, glog, and in that case you'll need to
+look down at the next section.
+
+Note: the official go policy is that libraries should not vendor their
+dependencies. This is unworkable for us, since our dependencies change and HEAD
+on every dependency has not necessarily been tested with client-go. In fact,
+HEAD from all dependencies may not even compile with client-go!
+
+## Dependency management for the serious (or reluctant) user
+
+Reasons why you might need to use a dependency management system:
+* You use a dependency that client-go also uses, and don't want two copies of
+  the dependency compiled into your application. For some dependencies with
+  singletons or global inits (e.g. `glog`) this wouldn't even compile...
+* You want to lock in a particular version (so you don't have to change your
+  code every time we change a public interface).
+* You want your install to be reproducible. For example, for your CI system or
+  for new team members.
+
+There are three tools you could in theory use for this. Instructions
+for each follows.
+
+### Dep
+
+[dep](github.com/golang/dep) is an up-and-coming dependency management tool,
+which has the goal of being accepted as part of the standard go toolchain. It
+is currently pre-alpha. However, it comes the closest to working easily out of
+the box.
+
+```sh
+$ go get github.com/golang/dep
+$ go install github.com/golang/dep/cmd
+
+# Make sure you have a go file in your directory which imports k8s.io/client-go
+# first--I suggest copying one of the examples.
+$ dep ensure k8s.io/client-go@^1.5.1
+```
+
+This will set up a /vendor directory in your current directory, add `k8s.io/client-go`
+to it, and flatten all of `k8s.io/client-go`'s dependencies into that vendor directory,
+so that your code and `client-go` will both get the same copy of each
+dependency.
+
+After installing like this, you could either use dep for your other
+dependencies, or copy everything in the `vendor` directory into your
+`$GOPATH/src` directory and proceed as if you had done a fancy `go get` that
+flattened dependencies sanely.
+
+One thing to note about dep is that it will omit dependencies that aren't
+actually used, and some dependencies of `client-go` are used only if you import
+one of the plugins (for example, the auth plugins). So you may need to run `dep
```

Comment:
For some reason we statically import the plugins: https://github.com/kubernetes/kubernetes/blob/6bfe0b6741d10574a1ab0b5a401b81831801d0a7/staging/src/k8s.io/client-go/kubernetes/clientset.go#L39-L39

Is this intentional?